samples/MvcCoreApp/Program.cs

```diff
@@ -1,9 +1,27 @@
 using System.Security.Claims;
+using Microsoft.AspNetCore.DataProtection;
+using Microsoft.AspNetCore.Identity;
 using Microsoft.AspNetCore.SystemWebAdapters;
 
 var builder = WebApplication.CreateBuilder();
 builder.Services.AddReverseProxy().LoadFromConfig(builder.Configuration.GetSection("ReverseProxy"));
 
+// These must match the data protection settings in MvcApp Startup.Auth.cs for cookie sharing to work
+builder.Services.AddDataProtection()
+    .PersistKeysToFileSystem(new DirectoryInfo("C:\\keyDirectory"))
```

Comment:
Try to figure out how to make this relative

Reply:
It would have to be relative to something that's available to both the ASP.NET and ASP.NET Core app, so being relative to either of them may not be very useful. I could imagine using `Path.Combine(Path.GetTempPath(), ApplicationName)` or something like that as an absolute path that both apps can find easily.

Reply:
Also, if we add extension methods for the ASP.NET side, we could add one for ASP.NET Core, too, that takes care of setting up data protection. That might not be as important, though, since the changes here are pretty minimal and I don't think we need to put the `AddAuthentication().AddCookie` call in adapters since that's code that will live on in the ASP.NET Core app even after migration is complete.

Reply:
At least include an app subfolder like:
```suggestion
    .PersistKeysToFileSystem(new DirectoryInfo("C:\\keyDirectory\MvcApp"))
```

Reply:
I went with `Path.Combine(Path.GetTempPath(), "sharedkeys", "mvcapp")` which seems to work well enough